alchemist/alchemist-fxui/src/main/java/it/unibo/alchemist/boundary/interfaces/FXOutputMonitor.java

```diff
@@ -0,0 +1,125 @@
+     * real time, slowing down the simulator if needed. If the simulation is
+     * slower than the real time, then the display refreshes fast enough to keep
+     * the default frame rate.
+     *
+     * @param realTime true for the real time mode
+     */
+    void setRealTime(boolean realTime);
+
+    /**
+     * Repaints this {@link javafx.scene.canvas.Canvas}' {@link javafx.scene.canvas.GraphicsContext}
+     * by drawing all the {@link it.unibo.alchemist.boundary.gui.effects.EffectFX Effect}s of each
+     * {@link it.unibo.alchemist.model.interfaces.Node} of the specified
+     * {@link it.unibo.alchemist.model.interfaces.Environment}.
+     */
+    void repaint();
+
+    /**
+     * Getter method for the {@link it.unibo.alchemist.boundary.gui.effects.EffectFX Effects} to draw.
+     *
+     * @return the current {@code Effects} to draw
+     */
+    Collection<EffectGroup<P>> getEffects();
+
+    /**
+     * Setter method for the effects to draw.
+     * <p>
+     * All previous set {@link it.unibo.alchemist.boundary.gui.effects.EffectFX Effects} are removed.
+     *
+     * @param effects the {@code Effects} to draw
+     */
+    void setEffects(Collection<EffectGroup<P>> effects);
+
+    /**
+     * Add all the {@link EffectGroup}s in the collection to the
+     * {@link it.unibo.alchemist.boundary.gui.effects.EffectFX Effects} to draw.
+     *
+     * @param effects the {@link EffectGroup}s to draw
+     * @see Collection#addAll(Collection)
+     */
+    void addEffects(Collection<EffectGroup<P>> effects);
+
+    /**
+     * Add the {@link EffectGroup} in the collection to the
+     * {@link it.unibo.alchemist.boundary.gui.effects.EffectFX Effects} to draw.
+     *
+     * @param effects the {@link EffectGroup} to draw
+     * @see Collection#add(Object)
+     */
+    void addEffectGroup(EffectGroup<P> effects);
+
+    /**
+     * Getter method for the current view status.
+     *
+     * @return the current {@code ViewStatus}
+     */
+    ViewStatus getViewStatus();
+
+    /**
+     * Setter method for the current view status.
+     *
+     * @param viewStatus the {@code ViewStatus} to set
+     */
+    void setViewStatus(ViewStatus viewStatus);
+
+    /**
+     * The enum models the status of the view.
+     */
+    enum ViewStatus {
+        /** In this status, click and drag to move the view. */
+        PANNING,
+        /** In this status, click and drag to select nodes. */
+        SELECTING,
+        /** In this status, click and drag to move selected nodes. */
+        MOVING,
+        /** In this status, click to clone nodes. */
+        CLONING,
+        /** In this status, click to delete nodes. */
+        DELETING
+    }
+
+    /**
+     * Returns the keyboard listener associated with this monitor.
+     * @return the listener
+     */
+    KeyboardActionListener getKeyboardListener();
+
+    /**
+     * Returns the JavaFX Node that is this monitor.
+     * @return the node
+     */
+    Node getNode();
```

Comment:
Does that mean this method returns `this` systematically?

Reply:
Yes, without this method a cast is necessary, keep it or use casts?

Reply:
refactor to `asJavaFXNode()`